osu.Game/Tests/Beatmaps/BeatmapConversionTest.cs

```diff
@@ -143,14 +143,19 @@ private ConvertResult read(string name)
             }
         }
 
-        private IBeatmap getBeatmap(string name)
+        public IBeatmap GetBeatmap(string name)
         {
             using (var resStream = openResource($"{resource_namespace}.{name}.osu"))
             using (var stream = new LineBufferedReader(resStream))
             {
                 var decoder = Decoder.GetDecoder<Beatmap>(stream);
                 ((LegacyBeatmapDecoder)decoder).ApplyOffsets = false;
-                return decoder.Decode(stream);
+                var beatmap = decoder.Decode(stream);
+
+                // not sure but seems to be required.
+                beatmap.BeatmapInfo.Ruleset = CreateRuleset().RulesetInfo;
```

Comment:
We won't want this. This test is typically used for arbitrary ruleset-to-ruleset conversion, this should fail the mania test at the very least (mania's expecting an osu! beatmap for its tests).

I think ideally we want this to be populated with the correct ruleset for the beatmap. But considering going forward and splitting out the rulesets into their own repos, how about:
1. If `BeatmapInfo.RulesetId == CreateRuleset().RulesetInfo.Id`, use the ruleset as you are here.
2. Otherwise, use `null`. This is typically used in places like `TaikoBeatmapConverter` and `ManiaBeatmapConverter` to check if the beatmap is for the current ruleset or not. `null` can be used to indicate that it isn't for the current ruleset.

Reply:
Hmm looks like this passed since the test sets its own `Ruleset`...

https://github.com/ppy/osu/blob/eee8af28f6756d6a5f52d9a3cb528bd0378451a2/osu.Game/Tests/Beatmaps/BeatmapConversionTest.cs#L104-L105